lib/service.js

```diff
@@ -20,10 +19,9 @@ class Service extends EventEmitter {
     this.initiated = false;
     this.stop = false;
     this.launchType = '';
-    this.eventBuffer = async.queue(this._eventBufferWorker.bind(this));
```

Comment:
it's a bit hard to figure out from the diff - where is this going?

Reply:
I merged _eventBufferWorker, _emitNewEvents, and _convertEvent. Rather than do that chain as a series of callbacks depending on async.queue to make it synchronous we just handle the events as they come in.